text/0635-mediaconvert-l2-construct.md

```diff
@@ -0,0 +1,258 @@
+# AWS MediaConvert L2 Constructs
+
+* **Original Author(s):**: @PavanRangudu
+* **Tracking Issue**: [#635](https://github.com/aws/aws-cdk-rfcs/issues/635)
+* **API Bar Raiser**: @{BAR_RAISER_USER}
+
+
+[AWS MediaConvert](https://aws.amazon.com/mediaconvert/) is a file-based video processing service that provides 
+scalable video processing for content owners and distributors with media libraries of any size. Today, customers can create
+MediaConvert JobTemplates, Presets, and Queues using autogenerated L1 (CloudFormation) Constructs. However, this involves 
+specifying detailed JSON structures as input that is cumbersome and error-prone. Developers need to understand the exact 
+format and structure required by MediaConvert, as well as how to translate that into CDK code. 
+
+This process could be greatly simplified with the support of L2 constructs. We will build convenience methods working backwards 
+from common use cases and default to recommended best practices.
+
+This RFC proposes L2 constructs for MediaConvert which will provide convenience features and abstractions for the existing 
+L1 (CloudFormation) Constructs building on the functionality already supported in the @aws-cdk/aws-mediaconvert module.
+
+
+## README
+
+AWS MediaConvert has three CloudFormation resources:
+
+- Job Templates:
+
+    Job Templates in AWS MediaConvert allow you to predefine the settings for your media processing jobs. By using Job Templates, you can tailor jobs to specific workflows, ensuring consistency and efficiency across your media processing tasks. These templates include settings for video and audio encoding, output formats, and more, simplifying the job creation process by reducing the need for manual configuration.
+
+- Presets:
+
+    Presets are reusable collections of settings that define how MediaConvert processes video, audio, and captions. Presets can be applied to different jobs, making it easy to standardize encoding settings across multiple tasks. They help streamline workflows by encapsulating complex encoding settings into a single, reusable entity.
+
+- Queues:
+
+    Queues in AWS MediaConvert are used to manage the order and priority of media jobs. By organizing jobs into different Queues, you can control the processing flow, optimize resource usage, and ensure that critical jobs are handled with the appropriate priority. Queues allow you to separate jobs by priority or workflow, helping manage costs and processing time effectively.
+
+
+### Customer Usage and Interaction
+
+#### Job Templates
+
+To create a Custom Job Template using an L2 Construct, you define the ``jobTemplateProps`` object, which includes all the 
+necessary settings for your media processing tasks. This object is then passed to the ``JobTemplate`` construct to create the custom template.
+
+    const jobTemplateProps: JobTemplateProps = {
```

Comment:
how can customers define a template, will they write it in JSON format ?

Reply:
Yes, they can write it in a JSON format. We also discussed about having some helper methods to produce this props object. I plan to add some more details on it in my next commit.